pkg/operation/botanist/component/coredns/coredns.go

```diff
@@ -800,9 +800,15 @@ func getSearchPathRewrites(enabled bool, clusterDomain string, commonSuffixes []
 	suffixRewrites := ""
 	for _, suffix := range commonSuffixes {
 		suffixRewrites = suffixRewrites + `
-  rewrite stop name regex (.*)` + regexp.QuoteMeta(suffix) + `\.(.+)\.svc\.` + quotedClusterDomain + ` {1}` + suffix
+  rewrite stop {
+    name regex (.*)\.` + regexp.QuoteMeta(suffix) + `\.svc\.` + quotedClusterDomain + ` {1}.` + suffix + `
+    answer name (.*)\.` + regexp.QuoteMeta(suffix) + ` {1}.` + suffix + `.svc.` +clusterDomain + `
+  }`
 	}
-	return `
-  rewrite stop name regex (.+)\.svc\.` + quotedClusterDomain + `\.(.+)\.svc\.` + quotedClusterDomain + ` {1}.svc.` + clusterDomain + `
-  rewrite stop name regex (.+)\.(.+)\.svc\.(.+)\.svc\.` + quotedClusterDomain + ` {1}.{2}.svc.` + clusterDomain + suffixRewrites
+	return suffixRewrites + `
+  rewrite stop {
+    name regex ([^\.]+)\.([^\.]+)\.svc\.` + quotedClusterDomain + `\.([^\.]+)\.svc\.` + quotedClusterDomain + ` {1}.{2}.svc.` + clusterDomain + `
+    answer name ([^\.]+)\.([^\.]+)\.svc\.` + quotedClusterDomain + ` {1}.{2}.svc.` + clusterDomain + `.{2}.svc.` + clusterDomain + `
```

Comment:
The source and target namespace do not need to be equal. Maybe it is better to use the same approach as with the common suffices.